bin/main.ml

```diff
@@ -12,18 +12,29 @@ module Input_format = struct
   let all = [ JSONSchema; OpenAPI ]
 end
 
-let generate_atd input_format path_in =
-  let ic = open_in path_in in
-  let input_content = really_input_string ic (in_channel_length ic) in
-  close_in ic;
-
+let generate_atd input_format paths =
   let generate =
     match input_format with
     | Input_format.JSONSchema -> Generator.make_atd_of_jsonschema
     | OpenAPI -> Generator.make_atd_of_openapi
   in
-  input_content |> generate |> print_string;
-  ()
+  print_endline (Generator.base (String.concat " " (List.map Filename.basename paths)));
+  let root =
+    match paths with
+    | [ _ ] -> `Default
+    | _ -> `Per_file
+  in
+  List.iter
+    (fun path ->
+      let root =
+        match root with
+        | `Default -> None
+        | `Per_file -> Some (path |> Filename.basename |> Filename.remove_extension |> Utils.sanitize_name)
```

Comment:
minor: wondering if it does make sense to use filename for root time always (even if only one input scheme file passed)
I used `jsonscheme2atd` for openapi files only, so I didn't have issues with root type name. @Khady if you think it is handy, feel free to change the default `root` type to filename

Reply:
Also it would simplify code (we can get rid of `Default` and `Per_file` variants)